commit dca24b633c829940ada7c6b0dd74558b54e39ece
Author: Konrad Dybcio <konradybcio@kernel.org>
Date:   Fri Jun 16 01:36:58 2023 +0200

    media: venus: core: Set up secure memory ranges for SC7180
    
    Not all SC7180 devices ship with ChromeOS firmware. WoA devices use
    Android-like TZ, which uses PAS for image authentication. That requires
    the predefined virtual address ranges to be passed via scm calls.
    Define them to enable Venus on non-CrOS SC7180 devices.
    
    Signed-off-by: Konrad Dybcio <konrad.dybcio@linaro.org>
    Signed-off-by: Stanimir Varbanov <stanimir.k.varbanov@gmail.com>
    Signed-off-by: Hans Verkuil <hverkuil-cisco@xs4all.nl>

diff --git a/drivers/media/platform/qcom/venus/core.c b/drivers/media/platform/qcom/venus/core.c
index 8972845569c2..d5e3dabc2e26 100644
--- a/drivers/media/platform/qcom/venus/core.c
+++ b/drivers/media/platform/qcom/venus/core.c
@@ -764,6 +764,10 @@ static const struct venus_resources sc7180_res = {
 	.vmem_size = 0,
 	.vmem_addr = 0,
 	.dma_mask = 0xe0000000 - 1,
+	.cp_start = 0,
+	.cp_size = 0x70800000,
+	.cp_nonpixel_start = 0x1000000,
+	.cp_nonpixel_size = 0x24800000,
 	.fwname = "qcom/venus-5.4/venus.mdt",
 };